library/alloc/src/vec/mod.rs

```diff
@@ -1001,6 +1001,22 @@ impl<T, A: Allocator> Vec<T, A> {
         }
     }
 
+    /// Converts the vector into [`Box<T>`][owned slice], assuming that
+    /// `self.len() == self.capacity()`.
+    ///
+    /// # Safety
+    /// It's undefined behavior to call this function if `self.capacity() > self.len()`.
```

Comment:
This should be updated to match #99790 and say that `len` must [fit](https://doc.rust-lang.org/std/alloc/trait.Allocator.html#memory-fitting) the allocation.

Reply:
I think I’ll check the documentation for `vec` module in another PR. New implementation doesn’t use `Vec<_>` at all.